File: nomad-samples/admin/py/basic-live-channel-1.1/schedule_event/add_input_schedule_event.py
from constants.system_constants import *
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def add_input_schedule_event(AUTH_TOKEN, CHANNEL_ID, INPUT_ID, ON_AIR_TIME):
    # Create header for the request
    HEADERS = {
        'Content-Type': 'application/json',
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    # Build the payload BODY
    BODY = {
        "channelId": CHANNEL_ID,
        "fixedOnAirTimeUtc": ON_AIR_TIME,
        "type": {
            "id": LIVE_INPUT_LOOKUP_ID,
            "description": "Live Input"
        },
        "liveInput": {
            "id": INPUT_ID,
            "description": "name"
        }
    }


    try:
        # Send the request
        RESPONSE = requests.post(ADMIN_URL + "/liveChannel/" + CHANNEL_ID + "/liveScheduleEvent",  headers= HEADERS, data= json.dumps(BODY))
    
        if not RESPONSE.ok:
            raise Exception()
        
        # Return JSON response
        return json.loads(RESPONSE.text)

    except:
        api_exception_handler(RESPONSE, "Add Input Schedule Event failed")

File: nomad-samples/admin/js/advanced-live-channel-1.1/schedule-event/add-input-schedule-event.js
import * as sysConstants from "../constants/system-constants.js";
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

/**
 * Add Input Schedule Event
 *
 * @param {string} authToken    | Authorization Token
 * @param {Object} data         | Body
 *
 * @returns JSON Object
 */
export default async function addInputScheduleEvent(authToken, data) {
    // Check for valid parameters
    if (!authToken || !data) {
        throw new Error("Add Input Schedule Event: Invalid API call");
    }

    // Build the payload body
    const body = {
        channelId: data.channelId,
        type: {
            lookupId: sysConstants.LIVE_INPUT_LOOKUP_ID,
            description: "Live Input"
        },
        liveInput: {
            lookupId: data.inputId,
            description: data.name
        },
        previousId: data.previousId
    };

    // Create header for the request
    const headers = new Headers();
    headers.append("Content-Type", "application/json");
    headers.append("Authorization", `Bearer ${authToken}`);

    // Send the request
    const response = await fetch(`${prjConstants.SERVER_URL}/LiveChannel/${data.channelId}/liveScheduleEvent`, {
        method: "POST",
        headers: headers,
        body: JSON.stringify(body)
    });

    // Check for success
    if (response && response.ok) {
        // Return JSON response
        return await response.json();
    }

    await apiExceptionHandler(response, "Add Input Schedule Event failed");
}
